pkg/agent/config/config_linux.go

```diff
@@ -22,20 +23,36 @@ func applyCRIDockerdAddress(nodeConfig *config.Node) {
 	nodeConfig.CRIDockerd.Address = "unix:///run/k3s/cri-dockerd/cri-dockerd.sock"
 }
 
-func applyContainerdQoSClassConfigFileIfPresent(envInfo *cmds.Agent, nodeConfig *config.Node) {
-	blockioPath := filepath.Join(envInfo.DataDir, "agent", "etc", "containerd", "blockio_config.yaml")
+// applyContainerdQoSClassConfigFileIfPresent checks if QoS class configuration
+// files exist in the filesystem. If so, their paths are included in containerdConfig.
+func applyContainerdQoSClassConfigFileIfPresent(root fs.FS, dataDir string, containerdConfig *config.Containerd) {
+	relaiveDataPath := strings.TrimPrefix(dataDir, string(os.PathSeparator))
+	blockioPath := filepath.Join(relaiveDataPath, "agent", "etc", "containerd", "blockio_config.yaml")
+
+	pathPrefix := ""
+	if filepath.IsAbs(dataDir) {
+		pathPrefix = string(os.PathSeparator)
+	}
 
 	// Set containerd config if file exists
-	if _, err := os.Stat(blockioPath); !errors.Is(err, os.ErrNotExist) {
-		logrus.Infof("BlockIO configuration file found")
-		nodeConfig.Containerd.BlockIOConfig = blockioPath
+	if info, err := fs.Stat(root, blockioPath); !errors.Is(err, fs.ErrNotExist) {
```

Comment:
better count on err and log all cases instead of  !errors.Is(err, os.ErrNotExist) ? 
```
if errors.Is(err, fs.ErrNotExist) { logrus(whatever) }
if err != nil ( logrus whatever )
info.IsDir() { logrus whatever ) 
......

```

Reply:
Gave it a bit of an overhaul, not sure it's any better.